### PR TITLE
Task/cx 6548 web automation implement tagging

### PR DESCRIPTION
**Pull request by SherBhachuOnFido**
# Problem
- There was previously no way of running/disabling tests based on any tagging criteria, which lead to inflexible test runs.

# Solution
-Implement a way to choose to run or not run certain tests based on tags which would give us greater flexibility going forward.

## Checklist

_put `n/a` if item is not relevant to PR changes_

- [x] Has the CHANGELOG been updated?
- [ ] Has the README been updated?
- [ ] Has the CONTRIBUTING doc been updated?
- [ ] Has the RELEASE_GUIDELINES been updated?
- [x] Has the TESTING_GUIDELINES been updated?
- [ ] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [ ] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [ ] Have any new automated tests been implemented or the existing ones changed?
- [ ] Have any new manual tests been written down or the existing ones changed?
- [ ] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

File: test/main.js [hunk 1]
```diff
@@ -165,10 +165,22 @@ const createMocha = (driver, testCase) => {
...
   // Since we want to execute the same tests against many browsers we need to prevent this behaviour by
   // clearing the require cache.
+  if (process.env.MOCHA_GREP) {
+    console.log(
+      `process.env.MOCHA_GREP is set, so running tests with the tags ${process.env.MOCHA_GREP}`
+    )
+  }
+  if (process.env.MOCHA_INVERT) {
+    console.log(
+      `process.env.MOCHA_INVERT is set, so not running tests with the tags ${process.env.MOCHA_INVERTO}`
```
Comment (kopijunkie) [inline test/main.js:180]: I think this should've been `process.env.MOCHA_INVERT`
```suggestion
      `process.env.MOCHA_INVERT is set, so not running tests with the tags ${process.env.MOCHA_INVERT}`
```

File: test/TESTING_GUIDELINES.md [hunk 2]
```diff
@@ -55,6 +55,17 @@ Our UI tests also run on BrowserStack. If you would like to run them on your Bro
 - Specific to Safari on BrowserStack, you need to explicitly set the `localhostUrl` value in `config.json` to
   be read as `"localhostUrl": "https://bs-local.com:8080/",` this is due to the way Safari handles `https://localhost`.
 
+### Running tests with or without specific tags
+
+- We can now run tests that contain, or do not contain certain tags in the following way...
+
+`MOCHA_GREP="@1|@2" npm run test:ui` will run tests that ONLY have the tag @1 OR @2.
```
Comment (kopijunkie) [inline test/TESTING_GUIDELINES.md:62]: Just so I know I understood the usage correctly, this mean that if for example I added a tag of `@test_here` and `@test_here_also` on a couple of UI tests in the Document scenario's test description & I run `MOCHA_GREP="@test_here|@test_here_also" npm run test:ui` that would work also & it's not dependent on just a bunch of number-based tags like in the guidelines doc here?

**Review by kopijunkie — CHANGES_REQUESTED** [review]

File: test/TESTING_GUIDELINES.md (hunk 2 above)
Comment (SherBhachuOnFido) [reply test/TESTING_GUIDELINES.md:62]: Yes, I just used the number based tags as an example. I'd suggest as part of this ticket, give it a try as per the acceptance criteria.

File: test/TESTING_GUIDELINES.md [hunk 3]
```diff
@@ -55,6 +55,17 @@ Our UI tests also run on BrowserStack. If you would like to run them on your Bro
...
+`MOCHA_GREP="@3" npm run test:ui` will run tests that ONLY have the tag @3.
+`MOCHA_GREP='@1.*@5' npm run test:ui`will run tests that ONLY have the tags @1 AND @5.
+`MOCHA_INVERT="@3" npm run test:ui` will NOT run tests that have the tag @3.
+`MOCHA_INVERT="@1|@2" npm run test:ui` will NOT run tests that have the tags @1 OR @2.
+`MOCHA_INVERT='@1.*@3' npm run test:ui` will NOT run tests that have the tags @1 AND @3.
```
Comment (stefaniacardenas) [inline test/TESTING_GUIDELINES.md:67]: Are we using these tags somewhere in the code or are these for future usage?

**Review by stefaniacardenas — APPROVED** [review]

File: test/TESTING_GUIDELINES.md (hunk 3 above)
Comment (SherBhachuOnFido) [reply test/TESTING_GUIDELINES.md:67]: Sorry, I'm OOO at the moment, but we will be using them in the future. I believe there was a PR merged recently for use on Percy, with the tag @percy

**Review by kopijunkie — APPROVED** [review]

**Pull request merged**